packages/odyssey-react-mui/src/DataTable/DataTable.tsx

```diff
@@ -0,0 +1,488 @@
+} from "./components/DataTableRowActions";
+import { useRowReordering } from "./hooks/useRowReordering";
+import { DataTableSettings } from "./components/DataTableSettings";
+
+export type DataTableProps = {
+  /**
+   * The columns that make up the table
+   */
+  columns: MRT_ColumnDef<MRT_RowData>[];
+  /**
+   * The data that goes into the table, which will be displayed
+   * as the table rows
+   */
+  data: MRT_TableOptions<MRT_RowData>["data"];
+  /**
+   * The total number of rows in the table. Optional, because it's sometimes impossible
+   * to calculate. Used in table pagination to know when to disable the "next"/"more" button.
+   */
+  totalRows?: number;
+  /**
+   * The function to get the ID of a row
+   */
+  getRowId?: MRT_TableOptions<MRT_RowData>["getRowId"];
+  /**
+   * The initial density of the table. This is available even if the table density
+   * isn't changeable.
+   */
+  initialDensity?: (typeof densityValues)[number];
+  /**
+   * If true, the end user will be able to change the table density.
+   */
+  hasChangeableDensity?: boolean;
+  /**
+   * If true, the end user can resize individual columns.
+   */
+  hasColumnResizing?: boolean;
+  /**
+   * If true, the end user will be able to show/hide columns.
+   */
+  hasColumnVisibility?: boolean;
+  /**
+   * If true, the end user will be able to filter columns.
+   */
+  hasFilters?: boolean;
+  /**
+   * If true, the table will include pagination controls.
+   */
+  hasPagination?: boolean;
+  /**
+   * If true, the table will include checkboxes on each row, enabling
+   * the user to select some or all rows.
+   */
+  hasRowSelection?: boolean;
+  /**
+   * If true, the global table search controls will be shown.
+   */
+  hasSearch?: boolean;
+  /**
+   * If true, the end user can sort columns (ascending, descending, or neither)
+   */
+  hasSorting?: boolean;
+  /**
+   * If true, the end user can reorder rows via a drag-and-drop interface
+   */
+  hasRowReordering?: boolean;
+  /**
+   * If true, the search field will include a Search button, rather than
+   * firing on input change.
+   */
+  hasSearchSubmitButton?: boolean;
+  /**
+   * The debounce time, in milliseconds, for the search input firing
+   * `onChangeSearch` when changed. If `hasSearchSubmitButton` is true,
+   * this doesn't do anything.
+   */
+  searchDelayTime?: number;
+  /**
+   * Callback that fires when a row (or rows) is selected or unselected.
+   */
+  onChangeRowSelection?: (rowSelection: MRT_RowSelectionState) => void;
+  /**
+   * Callback that fires whenever the table needs to fetch new data, due to changes in
+   * page, results per page, search input, filters, or sorting
+   */
+  onChangeQuery: ({
+    page,
+    resultsPerPage,
+    search,
+    filters,
+    sort,
+  }: {
+    page?: number;
+    resultsPerPage?: number;
+    search?: string;
+    filters?: DataFilter[];
+    sort?: MRT_SortingState;
+  }) => MRT_TableOptions<MRT_RowData>["data"];
```

Comment:
If these `MRT_*` types are not re-exported, it will be very hard for folks to write type-compliant code (without importing them from `material-react-table` directly, which has its own problems).

The prior version had this: 
```
export type {
  MRT_ColumnFiltersState,
  MRT_SortingState,
  MRT_ColumnDef as TableColumn,
} from "material-react-table";
```

I'd encourage you to add this back in and include any `MRT_*` types that are used in the various inputs here (I think, for example, that L54 will be hard to type properly without having `MRT_RowData`).